### PR TITLE
Add tooling documentation for Tools.Common, StaticAnalysis, and CI Filter

### PR DESCRIPTION
**Pull request by cormacpayne**
## Checklist

- [x] I have read the [_Submitting Changes_](../CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [ ] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [x] The PR does not introduce [breaking changes](../documentation/breaking-changes/breaking-changes-definition.md)
- [ ] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
    - [ ] the markdown help files have been regenerated using the commands listed [here](../documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

**Comment by cormacpayne** [comment]
@markcowl @MiYanni @niander would you mind taking a look and reviewing this? Let me know if there's anything I missed for any of these documents.

CC: @erich-wang @isra-fel @dolauli please have a look as well and let me know if you have any questions about the documents.

**Comment by isra-fel** [comment]
@msJinLei @wyunchi-ms please also take a look at this, thanks

File: documentation/tooling/ci-filter.md [hunk 1]
```diff
@@ -0,0 +1,81 @@
+## CI Filter
+
+### What is the CI Filter?
```
Comment (markcowl) [inline documentation/tooling/ci-filter.md]: Thanks, I didn't even think about this, but this is important to know.

File: documentation/tooling/ci-filter.md [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+The CI filter is a set of tooling that looks at the files changed in a pull request and makes decisions about what processes should occurr during the pull request's CI. These decisions include the following:
+
+- Which projects to build
+- Which tests to run
+- Which modules to generate help for
+- Which modules to run static analysis on
+
+This filtering allows for faster build times as we are no longer building _every_ project, running _every_ test, generating _all_ help and running static analysis on _everything_, but rather only on the services that are being updated in the pull request and what they are dependent on.
```
Comment (markcowl) [inline documentation/tooling/ci-filter.md]:
```suggestion
This filtering allows for faster build times as we are no longer building _every_ project, running _every_ test, generating _all_ help and running static analysis on _everything_, but rather only on the services that are being updated in the pull request and the services that depend on them.
```

**Pull request merged**